### PR TITLE
Fixes rendering problems with Question components.

### PR DESCRIPTION
**Pull request by sophiabersani**
Stories:
https://www.pivotaltracker.com/story/show/127104545
https://www.pivotaltracker.com/story/show/127931277
- Implements the new KeyboardAvoidingView to prevent question inputs from being overlapped by the software keyboards.
- Fixes Date type questions not rendering on iOS.
- Fixes a few warnings related to mismatching styling properties.

### CODE REVIEW

File: js/components/QuestionTypes/LongAnswerQuestion.js [hunk 1]
```diff
@@ -70,7 +70,7 @@ const LongAnswerQuestion = React.createClass({
           textStyle={Styles.question.headerText}>
             Question #{this.props.index}
         </ViewText>
-        <Text style={[Styles.type.h3, Styles.question.text]}>{this.props.text}</Text>
+        <Text style={[Styles.type.h3, Styles.question.text]}>{this.props.text}{this.props.text}{this.props.text}{this.props.text}{this.props.text}{this.props.text}</Text>
```
Comment (dan-nyanko) [inline js/components/QuestionTypes/LongAnswerQuestion.js]: Is this left-over from a test?

Comment (sophiabersani) [inline js/components/QuestionTypes/LongAnswerQuestion.js]: Please refresh your local repo, I believe git gave you an old commit.

**Comment by dan-nyanko** [comment]
@gbersani not sure if this is new or not; but within a long question input 
- on Android the soft keyboard 'return' key will insert a newline character
- on iOS the soft keyboard 'return' key exists the input

**Comment by dan-nyanko** [comment]
I've implemented similar behavior with the following [mixin](https://github.com/ecohealthalliance/GoodQuestion/blob/master/js/mixins/event-mixins.js#L48-L79)  

Please provide an explanation of why you choose to implement this 3rd party library along side the existing methodology.

**Comment by sophiabersani** [comment]
This is an official RN component that was not available when that mixin was first introduced.

The mixing worked well for most inputs, but in the current implementation we'd still lose about half of LongAnswerQuestion 's textboxes if the question spanned for more than a few lines. This issue was very noticeable on smaller screens.

**Comment by dan-nyanko** [comment]
Then we should either add a chore to PT to migrate the mixin methodology to this new component or take care of it within this branch.  Let me know

**Comment by sophiabersani** [comment]
We could migrate the rest, the mixin seems to be having trouble working sometimes after the updates to RN (0.27+) even on the older components.
This is a very simple implementation to a common problem and I feel we should take advantage of it.

I would like to have a chore to retire the mixing and implement this component to other views if needed. It seems to be way more stable across different devices and builds.

**Pull request merged**